kvmagent/kvmagent/test/vmmeta_testsuite/local/test_file_metadata_handler.py

```
@@ -0,0 +1,436 @@
+        meta_path = os.path.join(self.tmpdir, 'nonexistent.vmmeta')
+        self.handler._do_cleanup(meta_path)
+
+    def _test_cleanup_raises_if_main_file_cannot_be_removed(self):
+        meta_path = self._touch('abc123.vmmeta')
+        os.chmod(self.tmpdir, 0o444)
+        try:
+            with self.assertRaises(Exception) as ctx:
+                self.handler._do_cleanup(meta_path)
+            self.assertIn('failed to cleanup metadata file', str(ctx.exception))
+        finally:
+            os.chmod(self.tmpdir, 0o755)
```

Ruff (0.15.7)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)